features/steps.py

```diff
@@ -1,25 +1,38 @@
 # -*- coding: utf-8 -*-
 from contextlib import closing
-import datetime
+from datetime import datetime
 import os
-from journal import INSERT_ENTRY
-from journal import connect_db
-from journal import DB_SCHEMA
 from lettuce import world
 from lettuce import step
 from lettuce import before
 from lettuce import after
 import markdown
+from journal import Entry
+from sqlalchemy.orm import sessionmaker
+from sqlalchemy import *
+from pyramid import testing
+import transaction
 
-TEST_DSN = 'dbname=test_learning_journal user=roberthaskell'
+TEST_DSN = 'postgresql://roberthaskell:@/test_learning_journal'
 
 
 @before.all
 def db():
-    """set up a database"""
-    settings = {'db': TEST_DSN}
-    init_db(settings)
-    world.settings = settings
+    engine = create_engine(TEST_DSN)
+    Session = sessionmaker(bind=engine)
+    world.engine = engine
+    world.session = Session()
+    meta = MetaData()
+    entries = Table(
+    'entries', meta,
+    Column('id', Integer, primary_key=True, autoincrement=True),
+    Column('title', Unicode(127), nullable=False),
+    Column('text', UnicodeText, nullable=False,),
+    Column(
+        'created', DateTime, nullable=False, default=datetime.today)
+    )
+    world.db = entries
+    entries.create(engine, checkfirst=True)
```

Comment:
I assume your "my tests are shit" covers this, but I don't think you need to go through all this work.  There are better ways to handle setup for testing with a database.